library/std/src/diagnostics.qs

```diff
@@ -149,5 +149,73 @@ namespace Microsoft.Quantum.Diagnostics {
         areEqual
     }
 
-    export DumpMachine, DumpRegister, CheckZero, CheckAllZero, Fact, CheckOperationsAreEqual;
+    /// # Summary
+    /// Starts counting the number of times the given operation is called. The counter is reset to zero.
+    ///
+    /// # Description
+    /// This operation allows you to count the number of times a given operation is called. As part of
+    /// starting teh counting, the counter is reset to zero, which may override existing counts for the same operation.
+    /// Counting is based on the specific specialization of the operation invoked, so `X` and `Adjoint X` are counted separately.
+    /// Likewise `Controlled X`, `CNOT`, and `CX` are independent operations that are counted separately.
+    ///
+    /// # Input
+    /// ## callable
+    /// The operation to be counted.
+    @Config(Unrestricted)
+    operation StartCountingOperation<'In, 'Out>(callable : 'In => 'Out) : Unit {
+        body intrinsic;
+    }
+
+    /// # Summary
+    /// Stops counting the number of times the given operation is called and returns the count. Returns -1
+    /// if the operation was not being counted.
+    ///
+    /// # Description
+    /// This operation allows you to stop counting the number of times a given operation is called and returns the count.
+    /// If the operation was not being counted, it returns -1.
+    ///
+    /// # Input
+    /// ## callable
+    /// The operation whose count will be returned.
+    /// # Output
+    /// The number of times the operation was called since the last call to `StartCountingOperation`.
```

Comment:
Same as in counting qubits, it would be good to have a Q# code snippet in API docs for this

Reply:
Added some samples and clarifications.